interlok-core/src/main/java/com/adaptris/core/management/config/SharedConnectionConfigurationChecker.java

```diff
@@ -0,0 +1,125 @@
+package com.adaptris.core.management.config;
+
+import java.io.InputStream;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import com.adaptris.core.management.BootstrapProperties;
+import com.adaptris.core.management.UnifiedBootstrap;
+
+public class SharedConnectionConfigurationChecker implements ConfigurationChecker {
+  
+  private static final String FRIENDLY_NAME = "Shared connection check.";
+  
+  private static final String DESCRIPTION = "This test will scan your configuration for shared connections, making sure each is used and referenced correctly.";
+  
+  private static final String XPATH_AVAILABLE_CONNECTIONS = "//shared-components/connections/*/unique-id";
+  
+  private static final String XPATH_REFERENCED_CONSUME_CONNECTIONS = "//consume-connection/lookup-name";
+
+  private static final String XPATH_REFERENCED_PRODUCE_CONNECTIONS = "//produce-connection/lookup-name";
```

Comment:
Needs an extra `//connection[@class='shared-connection']/lookup-name` ?? (my xpath isn't great) to cope with shared connections inside services?

Reply:
Not sure this is working see comment below